pegasus_wrapper/artifact.py

```diff
@@ -62,7 +62,7 @@ class Artifact(Protocol):
     """
 
     depends_on: ImmutableSet[DependencyNode]
-    locator: Optional[Locator]
+    locator: Optional[Locator] = attrib(validator=optional(instance_of(Locator)))
```

Comment:
Although the type is `Optional[Locator]`, `attrs` requires an `attrib` with an optional validator in order to allow None